run_common.go

```diff
@@ -1444,17 +1454,57 @@ func cleanableDestinationListFromMounts(mounts []spec.Mount) []string {
 }
 
 // runSetupRunMounts sets up mounts that exist only in this RUN, not in subsequent runs
-func (b *Builder) runSetupRunMounts(mounts []string, sources runMountInfo, idMaps IDMaps) ([]spec.Mount, *runMountArtifacts, error) {
-	mountTargets := make([]string, 0, 10)
+//
+// If this function succeeds, the caller must unlock runMountArtifacts.TargetLocks (when??)
+func (b *Builder) runSetupRunMounts(bundlePath string, mounts []string, sources runMountInfo, idMaps IDMaps) ([]spec.Mount, *runMountArtifacts, error) {
+	mountTargets := make([]string, 0, len(mounts))
 	tmpFiles := make([]string, 0, len(mounts))
-	mountImages := make([]string, 0, 10)
+	mountImages := make([]string, 0, len(mounts))
+	intermediateMounts := make([]string, 0, len(mounts))
 	finalMounts := make([]specs.Mount, 0, len(mounts))
 	agents := make([]*sshagent.AgentServer, 0, len(mounts))
-	sshCount := 0
 	defaultSSHSock := ""
 	tokens := []string{}
 	lockedTargets := []string{}
+	var overlayDirs []string
+	succeeded := false
```

Comment:
This isn't being set to `true` when returning successfully.

Reply:
I don't think we're returning true for the release-1.26 branch also here and below fwiw.

Reply:
Please make sure one way or the other.  I thought CI had caught those in https://github.com/containers/buildah/pull/5931, and they'd been resolved.

Reply:
I double checked the release-1.26 branch against `v1.16.8`, there are 7 `succeeded` variables added and all are set to true before returning success.

Reply:
Thanks for the double check, my eyeballs must have been crossed on that review.